src/Lykke.Job.BlockchainTransactionsHistoryDetector/Workflow/Processes/NewTransactionDetectionProcess.cs

```diff
@@ -0,0 +1,247 @@
+using Lykke.Job.BlockchainTransactionsHistoryDetector.Core.Settings.BlockchainsIntegration;
+using Lykke.Service.BlockchainApi.Client;
+using Lykke.Service.BlockchainApi.Client.Models;
+
+
+namespace Lykke.Job.BlockchainTransactionsHistoryDetector.Workflow.Processes
+{
+    [UsedImplicitly]
+    public class NewTransactionDetectionProcess : IProcess
+    {
+        private readonly OnDemandDataCache<BlockchainAsset> _assetsCache;
+        private readonly BlockchainsIntegrationSettings _blockchainsIntegrationSettings;
+        private readonly IObservableWalletsRepository _observableWalletsRepository;
+        private readonly ILog _log;
+        
+        private IDictionary<string, BlockchainApiClient> _blockchainApiClients;
+        private CancellationTokenSource _cancellationTokenSource;
+        private IEventPublisher _eventPublisher;
+        private Task _task;
+
+
+        public NewTransactionDetectionProcess(
+            BlockchainsIntegrationSettings blockchainsIntegrationSettings,
+            IObservableWalletsRepository observableWalletsRepository,
+            ILog log)
+        {
+            // ReSharper disable HeapView.ObjectAllocation.Evident
+            
+            _assetsCache = new OnDemandDataCache<BlockchainAsset>();
+            _blockchainsIntegrationSettings = blockchainsIntegrationSettings;
+            _observableWalletsRepository = observableWalletsRepository;
+            _log = log;
+            
+            // ReSharper enable HeapView.ObjectAllocation.Evident
+        }
+
+
+        public void Dispose()
+        {
+            _cancellationTokenSource?.Cancel();
+
+            try
+            { 
+                _task?.Wait();
+            }
+            catch (AggregateException)
+            {
+
+            }
+        }
+
+        public void Start(ICommandSender commandSender, IEventPublisher eventPublisher)
+        {
+            _cancellationTokenSource = new CancellationTokenSource();
+            _eventPublisher = eventPublisher;
+
+            _task = Task.Run(async () =>
+            {
+                var cancellationToken = _cancellationTokenSource.Token;
+
+                await GetClientsForSupportedBlockchainsAsync();
+                
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    await DetectNewTransactionsAsync(cancellationToken);
+                }
+
+            });
+        }
+
+        private async Task DetectNewTransactionsAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                string continuationToken = null;
+
+                do
+                {
+
+                    IEnumerable<ObservableWalletDto> wallets;
+
+                    (wallets, continuationToken) = await _observableWalletsRepository.GetAllAsync(_blockchainApiClients.Keys, 100, continuationToken);
+
+                    await Task.WhenAll(wallets.Select
```

Comment:
There a couple of things here:
1. If one blockchain API is unavailable due to some reasons, then service can't perform its task at all. That's not good. Issue in the one blockchain API shouldn't affect rest of them.
2. Here could be generated a pretty big number of parallel requests. This could reduce overall performance of the blockchain API. Most proper way to process massive data, that I know, is TPL Dataflow - https://docs.microsoft.com/en-us/dotnet/standard/parallel-programming/dataflow-task-parallel-library. Example in Lykke - https://github.com/LykkeCity/Lykke.Logs/blob/Logging-system-redesign/src/Lykke.Logs/Loggers/LykkeAzureTable/AzureTableLogPersistenceQueue.cs. 

My suggestion is:

1. Extract class, that process transaction of particular blockchain. This class have its own processing thread (TimerTrigger).
2. Use ActionBlock to controll max degree of parallelism per blockchain API